### PR TITLE
Simple JDK 11 fixes

### PR DESCRIPTION
**Pull request by jrudolph**
These are the small and and simple parts of fixes for JDK 11, split out from #25923 for easier digestion.

Refs #25733.

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-actor-tests/src/test/scala/akka/io/TcpIntegrationSpecSupport.scala [hunk 1]
```diff
@@ -11,10 +11,14 @@ import akka.actor.ActorRef
 import akka.io.Inet.SocketOption
 import akka.testkit.SocketUtil._
 import Tcp._
+import akka.actor.ActorSystem
 
 trait TcpIntegrationSpecSupport { _: AkkaSpec ⇒
 
-  class TestSetup(shouldBindServer: Boolean = true) {
+  class TestSetup(shouldBindServer: Boolean = true, runClientInExtraSystem: Boolean = true) {
+    val clientSystem =
```
Comment (chbatey) [inline akka-actor-tests/src/test/scala/akka/io/TcpIntegrationSpecSupport.scala]: Will need shutdown or will leak for sbt shell

File: project/Dependencies.scala [hunk 2]
```diff
@@ -153,7 +153,7 @@ object Dependencies {
   val actorTests = l ++= Seq(
     Test.junit, Test.scalatest.value, Test.commonsCodec, Test.commonsMath,
     Test.mockito, Test.scalacheck.value, Test.jimfs,
-    Test.dockerClient
+    Test.dockerClient, Provided.activation // dockerClient needs javax.activation.DataSource in JDK 11+
```
Comment (chbatey) [inline project/Dependencies.scala]: :+1:

File: akka-actor-tests/src/test/scala/akka/io/TcpIntegrationSpecSupport.scala (hunk 1 above)
Comment (jrudolph) [reply akka-actor-tests/src/test/scala/akka/io/TcpIntegrationSpecSupport.scala]: Thanks, good point, will fix.

Comment (jrudolph) [reply akka-actor-tests/src/test/scala/akka/io/TcpIntegrationSpecSupport.scala]: Now the clientSystem is terminated when the main system used for the server is terminated as well.

**Review by johanandren — APPROVED** [review]

**Review by raboof — APPROVED** [review]

**Pull request merged**